File: ion-clients/src/ion_clients/core/configuration/candles.py
from pydantic import BaseModel
from typing import Any, Literal, Optional


class BaseResponse(BaseModel):
    data: Any
    response_code: Literal[200]
    error_message: str


class BaseCandlesResponseModel(BaseResponse):

    pass

class BaseCandlesDataModel(BaseModel):
    date: str
    vol: int
    open: Optional[float]
    high: Optional[float]
    close: Optional[float]
    low: Optional[float]

File: ion-frontend/src/data/schema/candles.ts
export type FinnhubCandlesSchema = FinnhubCandlesEntrySchema[];

export interface FinnhubCandlesEntrySchema {
    open: number;
    high: number;
    low: number;
    close: number;
    date: number;
    volume: number;
    symbol: string;
}
